### PR TITLE
fix: disable verify email button 

### PR DESCRIPTION
**Pull request by Chisomchima**
this PR attempts to fix this [comment](https://dhis2.atlassian.net/browse/DHIS2-18374?focusedCommentId=205915) for the verify email button [ticket](https://dhis2.atlassian.net/browse/DHIS2-18374)

### CODE REVIEW

File: src/layout/VerifyEmail.component.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ const sendEmailVerificationMutation = {
     type: 'create',
 }
 
+const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i
```
Comment (Chisomchima) [inline src/layout/VerifyEmail.component.js:12]: this regex validation was gotten from the email validation from the [dhis2-ui library](https://github.dev/dhis2/ui/blob/69b3ec912bbe75cccef2640d2d7ab317bb117ae1/collections/forms/src/validators/email.js#L28-L29).

File: src/layout/VerifyEmail.component.js [hunk 2]
```diff
@@ -9,6 +9,9 @@ const sendEmailVerificationMutation = {
     type: 'create',
 }
 
+const emailRegex =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i
```
Comment (tomzemp) [inline src/layout/VerifyEmail.component.js:13]: Suggest we use the RegEx that is used for the field validation for consistency and then potentially update both the field/button validator in the context of updating the email update workflow.

If we use different RegExes, the email can potentially fail front-end validation, not be posted to the backend, and then the verify email button will be enabled, but send to the old email address

**Review by tomzemp — COMMENTED** [review]
Looks good, but I think we should use consistent RegEx

**Review by tomzemp — APPROVED** [review]
lgtm 🎉

**Pull request merged**